tests/ga/test_update.py

```diff
@@ -1591,14 +1594,14 @@ def test_it_should_delete_extension_events_directory_if_extension_telemetry_pipe
                         self.assertFalse(os.path.exists(ext_dir), "Extension directory {0} still exists!".format(ext_dir)) # pylint: disable=bad-indentation
 
     def test_it_should_retain_extension_events_directories_if_extension_telemetry_pipeline_enabled(self):
-
-            # Rerun update handler again with extension telemetry pipeline enabled to ensure we dont delete events directories
-            with self._setup_test_for_ext_event_dirs_retention() as (update_handler, expected_events_dirs): # pylint: disable=bad-indentation
-                update_handler.run(debug=True) # pylint: disable=bad-indentation
-                for ext_dir in expected_events_dirs: # pylint: disable=bad-indentation
-                    self.assertTrue(os.path.exists(ext_dir), "Extension directory {0} should exist!".format(ext_dir)) # pylint: disable=bad-indentation
+        # Rerun update handler again with extension telemetry pipeline enabled to ensure we dont delete events directories
+        with self._setup_test_for_ext_event_dirs_retention() as (update_handler, expected_events_dirs):
+            update_handler.run(debug=True)
+            for ext_dir in expected_events_dirs:
+                self.assertTrue(os.path.exists(ext_dir), "Extension directory {0} should exist!".format(ext_dir))
 
 
+@patch('azurelinuxagent.ga.update.get_collect_logs_handler')
```

Comment:
I thought it was an easy thing to do, but when adding these patches to the class setup, I ran into some issues like "call_count not present, return_value not present" for the patches. After about 30 minutes I gave up, since it's not crucial for this PR. Could be done in the future though.

Reply:
awesome thanks for trying!